src/test/scala/nl/biopet/bioconda/BiocondaRecipeTest.scala

```diff
@@ -0,0 +1,127 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package nl.biopet.bioconda
+
+import java.io.File
+
+import org.scalatest.Matchers
+import org.scalatest.testng.TestNGSuite
+import org.testng.annotations.Test
+
+import scala.io.Source
+class BiocondaRecipeTest extends TestNGSuite with Matchers {
+
+  val testRecipe = new BiocondaRecipe(
+    name = "test",
+    version = "1.0",
+    command = "testing",
+    sourceUrl = "test.example.com/test.jar",
+    sourceSha256 = "af15", //incorrect but that is irrelevant
+    runRequirements = Seq("openjdk"),
+    buildRequirements = Seq(),
+    testCommands = Seq("testing --version"),
+    homeUrl = "test.example.com/index.html",
+    license = "MIT",
+    summary = "test is a tool that is tested in this test suite.",
+    defaultJavaOptions = Seq("-Xms25m"),
+    buildNumber = 0,
+    notes = Some("This is java")
+  )
+
+  @Test
+  def testCreateRecipeFiles(): Unit = {
+    val tmp = File.createTempFile("recipe", "dir")
+    // The file should be a dir.
+    tmp.delete()
+    tmp.mkdir()
+    testRecipe.createRecipeFiles(tmp)
+    val metaYaml = new File(tmp, "meta.yaml")
+    val buildSh = new File(tmp, "build.sh")
+    val wrapper = new File(tmp, testRecipe.wrapperFilename)
+    metaYaml should exist
+    buildSh should exist
+    wrapper should exist
+    assert(wrapper.canExecute)
+    assert(!buildSh.canExecute)
+    assert(!metaYaml.canExecute)
+    Source.fromFile(metaYaml).mkString should equal(testRecipe.metaYaml)
+    Source.fromFile(buildSh).mkString should equal(testRecipe.buildScript)
+    Source.fromFile(wrapper).mkString should equal(testRecipe.wrapperScript)
+  }
+
+  @Test
+  def testRecipeValues(): Unit = {
+    testRecipe.fileName shouldBe ("test.jar")
+    testRecipe.wrapperFilename shouldBe ("test.py")
+  }
+
+  @Test
+  def testMetaYaml(): Unit = {
+    val yaml = testRecipe.metaYaml
```

Comment:
Please use a parser here to test, just grepping the content maybe fails after a reformat of the yaml files. CommonUtils has method for this

Reply:
Building a yaml parser is a crazy amount of work. I hoped to use the common-utils one, but that does not fly for some reason.

Reply:
Common-utils one is now used to create the parser.
Grepping the contents is not perfect, but it does test layout as well, which is a good thing. It should be a nice human-readable yaml.
If this breaks very often in testing I will change it. If not, let's keep it this way.

Reply:
:+1: